### PR TITLE
S3 project tentative merge

### PR DESCRIPTION
**Pull request by pleblanc1976**
This is as much to test buildbot changes as the merge.  I haven't tested the final merge of develop->S3-project myself yet; prob doesn't work with not-root + SKIP right now.  Don't merge until I give the all-clear.

(Holy moly this is a big PR)

### CODE REVIEW

**Comment by pleblanc1976** [comment]
looks like BB is not getting the libmarias3 submodule.  Need to hack on the cmakelists to make external-project do it, or make BB fetch it.

File: storage-manager/src/S3Storage.cpp [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+        { 
+            if (ms3_server_error(creds))
+                logger->log(LOG_WARNING, "S3Storage::getObject(): failed to GET, server says '%s'.  bucket = %s, key = %s."
+                    "  Retrying...", ms3_server_error(creds), bucket.c_str(), sourceKey.c_str());
+            else 
+                logger->log(LOG_WARNING, "S3Storage::getObject(): failed to GET, got '%s'.  bucket = %s, key = %s.  Retrying...", 
+                    s3err_msgs[err], bucket.c_str(), sourceKey.c_str());
+            sleep(5);
+        }
+    } while (err && retryable_error(err));
```
Comment (LinuxJedi) [inline storage-manager/src/S3Storage.cpp:237]: If S3 is down or there are WAN issues, won't this become an infinite loop? Maybe abort after X tries?

File: storage-manager/src/S3Storage.cpp [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    // no s3-s3 copy yet.  get & put for now.
+    
+    int err;
+    boost::shared_array<uint8_t> data;
+    size_t len;
+    err = getObject(sourceKey, &data, &len);
+    if (err)
+        return err;
+    return putObject(data, len, destKey);
+#endif
```
Comment (LinuxJedi) [inline storage-manager/src/S3Storage.cpp:422]: The section can probably go completely now?

File: versioning/BRM/dbrm.cpp [hunk 3]
```diff
@@ -3360,6 +3360,8 @@ vector<InlineLBIDRange> DBRM::getEMFreeListEntries() throw()
 
 int DBRM::takeSnapshot() throw ()
 {
+    return 0;   // don't know why, but we're calling this all the time.  Need to take most/all of those calls out, it's very wasteful.
```
Comment (LinuxJedi) [inline versioning/BRM/dbrm.cpp:3363]: Do we not need to do this at least sometimes?

**Review by LinuxJedi — APPROVED** [review]
Excellent work. Couldn't spot any obvious bad things. Left some comments.

**Pull request merged**